ARCHITECTURE.md

```diff
@@ -0,0 +1,56 @@
+# Architecture
+
+
+
+# Overview
+
+**`node-nikita-arch`** leverages our automation and deployment solution written in Nodejs, [Nikita](https://nikita.js.org/), to provide tools for automated installation of Arch Linux on a local or remote machine. Of note we provide here two main Nodejs modules, `bootstrap` and `system`, that encapsulate a set of Nikita actions to ***(i)*** setup disk partitions and mount them accordingly, ***(ii)*** install and configure mandatory system requirements and drivers and ***(iii)*** run services and install more specific softwares (e.g., Atom, Sublime Text, Java, ) based on user configuration.
+
+
+
+# Code map
+
+Herein we will describe the important modules and piece of code that articulate our repository. We will focus notably on how each interact with others to provide a comprehensive deployment and installation of Arch Linux.
+
+#### `lib/`
```

Comment:
Titles shall cover the functional scope, not the file names. A user expect to undestand from the title where is the source code, how the configuration is working, then you can tell him in the content that it is is which folder, which files are generated...

Reply:
I don't understand what you mean by *functional scope*.

Reply:
Take `./lib`, this is about where to locate the source code. Take `config.coffee`, this is about how does the configuration work. In the title, we dont care about the technical implementation, we care about the functionalities.

Reply:
For exemple, start with your user experience. As an end user, when I read the readme or when I run the command, what I know from nikita-arch is
1. It prompts me for questions, I might need to know why and how this is implemented if I see an issue with this
2. there are two main commands, I might need to know why and how this implemented if I want to report an issue, fix something or customize it
3. There is this concept of local versus remote, why and how is it implemented
4. ...

Reply:
Ok much clear :+1: